stellargraph/layer/sort_pooling.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright 2018-2020 Data61, CSIRO
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import tensorflow as tf
+from tensorflow.keras.layers import Layer
+from ..core.experimental import experimental
+
+
+@experimental(reason="Missing unit tests and generally untested.", issues=[1044])
+class SortPooling(Layer):
+
+    """
+    Sort Pooling Keras layer.
+
+    Note that sorting is performed using only the last column of the input tensor as stated in [1], "For convenience,
+    we set the last graph convolution to have one channel and only used this single channel for sorting."
+    
+    [1] An End-to-End Deep Learning Atchitecture for Graph Classification, M. Zhang, Z. Cui, M. Neumann, and
+    Y. Chen, AAAI-18, https://www.aaai.org/ocs/index.php/AAAI/AAAI18/paper/viewPaper/17146
+
+
+    Args:
+        k (int): The number of rows of output tensor.
+        flatten_output (bool): If True then the output tensor is reshaped to vector for each tensor in the batch.
```

Comment:
I thinking using "each tensor in the batch" is a bit misleading; because there's always only one output tensor, and this is actually referring to the batch elements within that overall tensor? Alternatives might be:

```suggestion
        flatten_output (bool): If True then the output tensor is reshaped to vector for each element in the batch.
```

or

```suggestion
        flatten_output (bool): If True then the output tensor is reshaped to vector for each row in the batch.
```